Sourcery/Parsing/FileParser.swift

```diff
@@ -594,8 +666,17 @@ extension FileParser {
             }
         }
 
+        var genericTypes: [Generic] = []
+        if let genericSource = extract(.name, from: source),
+            let parameterStartIndex = genericSource.index(of: "("),
+            let whereClauseSource = extract(.nameSuffixUpToBody, from: source) {
```

Comment:
this might not work with protocols where methods don't have body. we should probably extract it along with extracting return type (above), which handles this case